rero_mef/authorities/gnd/tasks.py

```diff
@@ -0,0 +1,250 @@
+                'verb': self.verb
+            }
+        if access_token:
+            params['accessToken'] = access_token
+
+        self.oai_response = self.sickle.harvest(**params)
+        error = self.oai_response.xml.find(
+            './/' + self.sickle.oai_namespace + 'error')
+        if error is not None:
+            code = error.attrib.get('code', 'UNKNOWN')
+            description = error.text or ''
+            try:
+                raise getattr(
+                    oaiexceptions, code[0].upper() + code[1:])(description)
+            except AttributeError:
+                raise oaiexceptions.OAIError(description)
+        if self.resumption_token:
+            # Test we got a complete response ('resumptionToken' in xml)
+            resumption_token_element = self.oai_response.xml.find(
+                './/' + self.sickle.oai_namespace + 'resumptionToken')
+
+            if resumption_token_element is None:
+                msg = ('ERROR HARVESTING incomplete response:'
+                       '{cursor} {token}').format(
+                    cursor=self.resumption_token.cursor,
+                    token=self.resumption_token.token
+                )
+                current_app.logger.error(msg)
+            else:
+                self.resumption_token = self._get_resumption_token()
+
+                self._items = self.oai_response.xml.iterfind(
+                    './/' + self.sickle.oai_namespace + self.element)
+
+        else:
+            # first time
+            self.resumption_token = self._get_resumption_token()
+
+            self._items = self.oai_response.xml.iterfind(
+                './/' + self.sickle.oai_namespace + self.element)
+
+
+@shared_task
+def process_records_from_dates(from_date=None, until_date=None,
+                               ignore_deleted=False, dbcommit=True,
+                               reindex=True, test_md5=True, verbose=False,
+                               **kwargs):
+    """Harvest multiple records from an OAI repo.
+
+    :param name: The name of the OAIHarvestConfig to use instead of passing
+                 specific parameters.
+    :param from_date: The lower bound date for the harvesting (optional).
+    :param until_date: The upper bound date for the harvesting (optional).
+    """
+    # data on GND Servers starts on 1900-01-01
+    name = 'gnd'
+    days_spann = 4
+    lastrun = None
+    url, metadata_prefix, lastrun, setspecs = get_info_by_oai_name(name)
+
+    request = Sickle(url, iterator=MyOAIItemIterator)
+
+    dates = {
+        'from': from_date or lastrun,
+        'until': until_date
+    }
+    update_last_run = from_date is None and until_date is None
+    # Sanity check
+    if (dates['until'] is not None) and (dates['from'] > dates['until']):
+        raise WrongDateCombination("'Until' date larger than 'from' date.")
+
+    lastrun_date = datetime.datetime.now()
+
+    setspecs = setspecs.split() or [None]
```

Comment:
Maybe a small comment to explain what you're using `[None]` instead of `[]`. (cfr L144-145).
For me you would enter, at least, one time in the for loop ?

Reply:
ok